libvast/include/vast/query_queue.hpp

```diff
@@ -53,6 +53,7 @@ class query_queue {
     uuid partition;
     uint64_t priority = 0;
```

Comment:
When calculating the priority of a query we should take into account whether a partition is erased or not. An erased partition should likely have a priority of 0.

Reply:
I'm not sure adding such an effect is a good idea. That would mean erasing a partition prolongs related queries while we already know there won't be any more results. I'd rather not touch the priority here.

Reply:
I can see both sides. Fine by me. Can you add a comment on why we don't touch the priority in the function that marks a partition erased for future reference?

Reply:
I don't think it makes sense to comment about why something is not done unless the reader would totally expect that thing. In this case a comment would be warranted if it would adjust the priority.

Also, a changing the priority here would require us to sort as well, otherwise the effect would be very unpredictable.

Reply:
If we would update priority, wouldn't it make more sense anyways to set it to "infinite" rather than zero? Since it costs nothing to evaluate an erased partition, we should do so as soon as possible.

Reply:
i thought about that as well, but let's weigh cost vs benefit.

The benefit is that a backlogged query that gets a candidate erased would be able to finish a bit faster _iff the erased partition is the one with the lowest prio of that query_.

The cost is that implementing a "set it to infinite" is not straight forward, because the current priority type is `uint64_t` which does not have a notion of "infinite", and wraps around. It becomes a game of tradeoffs, the options are:
1. Check for `erased` before updating the priority in all places - brittle
2. Use `double` instead - IEEE 754 is a mess but probably the sanest option
3. Choose a reasonably high value that is still higher than what you can reasonably expect from normal operation - also brittle, what if we need to scale the range at some point?
4. Implement a custom integral wrapper that prevents the wraparound - high effort and future maintenance

I'm going to close this discussion now because we've already spent too much time on this minor detail.